src/App.tsx

```diff
@@ -137,6 +138,12 @@ const App: FC = () => {
 
 export default () => (
   <AppSettingProvider>
-    <App />
+    {CLOUD && isFlagEnabled('multiOrg') && shouldUseQuartzIdentity() ? (
+      <UserAccountProvider>
+        <App />
+      </UserAccountProvider>
+    ) : (
+      <App />
+    )}
```

Comment:
My recommendation is to wait to figure this out until the other PR, which refactors the account context provider, is tested, merged, and working. I do think it's a good idea to move the Account Provider higher up, but we don't want to subordinate to feature flags. Otherwise, we'll break the account information for users if we have to flip off one of these flags.

Reply:
Let's take this one slow and think about it. But my initial thinking is that the appropriate check would be something like `if (CLOUD)` rather than `if (CLOUD)` plus feature flag logic.

Reply:
I spun up OSS instance and tested it with multiOrg on/off. Worked fine. Noticed one thing with Billing Page is that in OSS, it's redirected to 404 which wasn't happening for `accounts/settings` page. So, moved this route behind cloud flag as well so we get a consistent response.